### PR TITLE
PE-324: Implement move file command

### PR DESCRIPTION
**Pull request by fedellen**
This PR implements the move file command.

In order to use the complete meta data of a file for the move method, there was a refactor required in the ArFS file types. This refactor added all fields available from the data JSON on to the meta data. With this change, `file-info` will now display the complete meta data of a file. The `list` commands however, will continue to display just the combined data from the `fileOrFolder` type.

All of the get/info commands were previously displaying an extra field due as a result of the base types from core, `syncStatus`. With the above changes, they would also now display `lastModifiedData` regardless of whether the entity had that field.

Because of this, I added a quick workaround fix in the CLI layer to trim these fields from being displayed to the user. The fix included casting each of the result types as a Partial, so we could then delete any unwanted object fields.

### CODE REVIEW

File: src/ardrive.ts [hunk 1]
```diff
@@ -212,6 +217,101 @@ export class ArDrive extends ArDriveAnonymous {
...
+			stubUnixTime,
+			stubTransactionID,
+			stubDataContentType
+		);
+
+		const moveBaseCosts = await this.estimateAndAssertCostOfMoveFile(stubbedFileTransactionData);
+
+		const actualDriveId = baseFileMetaData.driveId;
+
+		if (driveId !== actualDriveId) {
```
Comment (arielmelendez) [inline src/ardrive.ts:237]: Check this earlier if possible

File: src/ardrive.ts [hunk 2]
```diff
@@ -212,6 +217,101 @@ export class ArDrive extends ArDriveAnonymous {
 		return driveId;
 	}
 
+	async movePublicFile(fileId: FileID, newParentFolderId: FolderID): Promise<ArFSResult> {
+		const driveId = await this.getDriveIdAndAssertDrive(newParentFolderId);
+
+		const baseFileMetaData = await this.getPublicFile(fileId);
```
Comment (arielmelendez) [inline src/ardrive.ts:223]: Rename to `origFileMetadata` or similar. Base is too overloaded a concept.

File: src/ardrive.ts [hunk 3]
```diff
@@ -212,6 +217,101 @@ export class ArDrive extends ArDriveAnonymous {
...
+			tips: [],
+			fees: {
+				[moveFileResult.metaDataTrxId]: +moveFileResult.metaDataTrxReward
+			}
+		});
+	}
+
+	async movePrivateFile(fileId: FileID, newParentFolderId: FolderID, driveKey: DriveKey): Promise<ArFSResult> {
+		const driveId = await this.getDriveIdAndAssertDrive(newParentFolderId, driveKey);
+		const baseFileMetaData = await this.getPrivateFile(fileId, driveKey);
```
[the remark this answers is not in the document]
Comment (arielmelendez) [reply src/ardrive.ts:268]: Remove all stubbies

File: src/utils/arfs_builders/arfs_file_builders.ts [hunk 4]
```diff
@@ -55,16 +55,17 @@ export class ArFSPublicFileBuilder extends ArFSFileBuilder<ArFSPublicFile> {
...
 			this.size = dataJSON.size;
 			this.lastModifiedDate = dataJSON.lastModifiedDate;
 			this.dataTxId = dataJSON.dataTxId;
 			this.dataContentType = dataJSON.dataContentType;
 
+			if (!this.name || !this.size || !this.lastModifiedDate || !this.dataTxId || !this.dataContentType) {
+				throw new Error('Invalid file state');
+			}
+
+			// TODO: WHY IS THE ArFSPublicFile type missing these fields?
```
Comment (arielmelendez) [inline src/utils/arfs_builders/arfs_file_builders.ts:67]: Remove me

Comment (fedellen) [reply src/utils/arfs_builders/arfs_file_builders.ts]: Included in 57c5925

File: src/ardrive.ts [hunk 5]
```diff
@@ -987,6 +1108,10 @@ export class ArDrive extends ArDriveAnonymous {
 		return this.arFsDao.getDriveIdForFolderId(folderId);
 	}
 
+	async stubDriveKey(): Promise<DriveKey> {
```
[the remark this answers is not in the document]
Comment (fedellen) [reply src/ardrive.ts]: Removed in 57c5925

File: src/utils/arfs_builders/arfs_builders.ts [hunk 6]
```diff
@@ -30,14 +30,16 @@ export abstract class ArFSMetadataEntityBuilder<T extends ArFSEntity> {
...
 	 *
-	 * @param tags (optional) a pre-fetched GQL node containing the txID and tags that will be parsed out of the on-chain data
+	 * @param node (optional) a pre-fetched GQL node containing the txID and tags that will be parsed out of the on-chain data
+	 *
+	 * @param owner (optional) filter all transactions out by owner's public arweave address
 	 *
 	 * @returns an array of unparsed tags
 	 */
-	protected async parseFromArweaveNode(node?: GQLNodeInterface): Promise<GQLTagInterface[]> {
+	protected async parseFromArweaveNode(node?: GQLNodeInterface, owner?: ArweaveAddress): Promise<GQLTagInterface[]> {
```
[the remark this answers is not in the document]
Comment (fedellen) [reply src/utils/arfs_builders/arfs_builders.ts:39]: https://ardrive.atlassian.net/browse/PE-524

**Pull request merged**